### ISSUE TITLE
Rename some transforms using appropriate wide adopted naming conventions

### ISSUE THREAD

**Issue by kirk86** [pytorch/vision#3941]
## 🚀 Feature

Rename confusing transformation names such as `transforms.Normalize()` to `transforms.Standardize()` and decouple combined operations `transforms.ToTensor()` into `transforms.Normalize()/transforms.Rescale()` and `transforms.ToTensor()` for clarity.

## Motivation

I'm always frustrated when trying to communicate the above operations (i.e. `transforms.Normalize()`, `transforms.ToTensor()`) to statisticians, the reason for that is that statisticians use the following conventional naming `standardize = rescale data to zero mean and unit variance` i.e., what `transforms.Normalize()` does, and, `normalize = rescale data value into the range[0, 1]`, i.e., what `tranforms.ToTensor()` does, which is basically 2 operations, transforming the PIL image into a tensor and rescaling it.

## Pitch

Is it possible to rename according to the world wide convention that statisticians use in order to avoid confusion in the future and have better clarity of what each operation is meant to be doing?

1. Rename `transforms.Normalize()` into `transforms.Standardize()`
2. Break `transforms.ToTensor` into two separate operations `transforms.Normalize()` which basically rescales the values into the range[0, 1], and `transforms.ToTensor()` which basically transforms a PIL image into a torch tensor.

**Comment by datumbox** [comment]
@kirk86 Thanks for the feedback.

The implicit scaling done by ToTensor is something has been raised by a few people in the past. Key problem with renaming them is that it's a massive breaking change that will likely affect all the pipelines that people wrote on top of TorchVision. I think your criticism is valid and worth taking on board when we refactor the transforms.

**Comment by NicolasHug** [comment]
Regarding Normalize -> Standardize: I agree.

For ref, in scikit-learn the "zero mean and unit variance" transformer is the `StandardScaler`, and the `Normalizer` just divides each input vector by its norm.
